plutus-tx/src/PlutusTx/Data/AssocMap.hs

```diff
@@ -389,23 +396,58 @@ toList d = go (toBuiltinList d)
             (P.unsafeFromBuiltinData (BI.fst hd), P.unsafeFromBuiltinData (BI.snd hd))
               : go tl
         )
-{-# INLINEABLE toList #-}
+{-# INLINEABLE toSOPList #-}
 
 -- | Convert the `Map` to a `P.BuiltinList` of key-value pairs. This operation is O(1).
 toBuiltinList :: Map k a -> BI.BuiltinList (BI.BuiltinPair BuiltinData BuiltinData)
-toBuiltinList (Map d) = d
+toBuiltinList = coerce
```

Comment:
Given that `Map` is a newtype and as such is representationally equal to its internals, does this change make any difference?